src/pcapi/serialization/spec_tree.py

```diff
@@ -38,3 +41,35 @@ def _generate_spec(self) -> dict:
                     schema["paths"][path][method.lower()]["security"] = func.requires_authentication
 
         return schema
+
+
+class ExtendedResponse(SpectreeResponse):
+    def __init__(
+        self,
+        *codes,
+        code_descriptions=None,
+        **code_models,
+    ):
+        self.code_descriptions = code_descriptions or {}
+        super().__init__(*codes, **code_models)
+
+    def get_code_description(self, code):
+        return self.code_descriptions[code] if code in self.code_descriptions else DEFAULT_CODE_DESC[code]
+
+    def generate_spec(self):
+        """
+        extend the spec to add custom descriptions of error codes
+
+        :returns: JSON
```

Comment:
(détail) ça renvoie plutôt un dictionnaire, non ?

```suggestion
        :returns: dict
```

Mais je ne sais pas si cette ligne de docstring est très utile. Ça n'est pas notre code qui va appeler cette méthode, donc probablement personne ne sera intéressé par connaître son type de retour. Au pire on l'ajoute en annotation de typing : `-> dict` ?

Reply:
oui j'ai copié collé un peu bêtement